apps/web/src/components/AssessmentStepTracker/utils.ts

```diff
@@ -451,3 +426,72 @@ export const groupPoolCandidatesByStep = (
 
   return stepsWithGroupedCandidates;
 };
+
+export type ResultFilters = {
+  query: string;
+  [NO_DECISION]: boolean;
+  [AssessmentDecision.Successful]: boolean;
+  [AssessmentDecision.Hold]: boolean;
+  [AssessmentDecision.Unsuccessful]: boolean;
+};
+
+export const defaultFilters: ResultFilters = {
+  query: "",
+  [NO_DECISION]: true,
+  [AssessmentDecision.Successful]: true,
+  [AssessmentDecision.Hold]: true,
+  [AssessmentDecision.Unsuccessful]: true,
+};
+
+const statusIsAvailable = (filter?: boolean) => {
+  return typeof filter !== "undefined" && filter;
+};
+
+export const filterResults = (
+  filters: ResultFilters,
+  steps: StepWithGroupedCandidates[],
+): StepWithGroupedCandidates[] => {
+  return steps.map((step) => {
+    const filteredResults = step.results.filter(
+      ({ poolCandidate: { user }, decision }) => {
+        if (filters.query) {
+          const fullName = [user.firstName, user.lastName]
+            .filter(notEmpty)
+            .join(" ");
+          if (!fullName.includes(filters.query)) {
+            return false;
+          }
+        }
+
+        let available: boolean = true;
+        switch (decision) {
+          case NO_DECISION:
+            available = statusIsAvailable(filters[NO_DECISION]);
+            break;
+          case AssessmentDecision.Hold:
+            available = statusIsAvailable(filters[AssessmentDecision.Hold]);
+            break;
+          case AssessmentDecision.Successful:
+            available = statusIsAvailable(
+              filters[AssessmentDecision.Successful],
+            );
+            break;
+          case AssessmentDecision.Unsuccessful:
+            available = statusIsAvailable(
+              filters[AssessmentDecision.Unsuccessful],
+            );
+            break;
+          default:
+            break;
+        }
+
+        return available;
```

Comment:
I haven't tested, but I'm pretty sure this is equivalent. It just needs some comments explaining why it works. Here's my first attempt.
```suggestion
        // `filters` shows which decisions we want to show.
        // If `filters.decision` is false, that we means we want to filter out all candidates with that decision, so they don't appear in results.
        return filters[decision];
```

Reply:
Oh yeah, that works as well 😅 I was trying to commit suggestion but since I touched that file it couldn't 👎 

Updated in [simplify status filter further](https://github.com/GCTC-NTGC/gc-digital-talent/pull/9158/commits/37cb3868cc1277d953379c3d81725f723a6eaf38)